leaderboard/redis/cluster_client.go

```diff
@@ -0,0 +1,208 @@
+func (cc *clusterClient) Ping(ctx context.Context) error {
+	err := cc.ClusterClient.Ping(ctx).Err()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SAdd call redis SADD function
+func (cc *clusterClient) SAdd(ctx context.Context, key, member string) error {
+	_, err := cc.ClusterClient.SAdd(ctx, key, member).Result()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// SRem call redis SREM function
+func (cc *clusterClient) SRem(ctx context.Context, key, member string) error {
+	err := cc.ClusterClient.SRem(ctx, key, member).Err()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TTL call redis TTL function
+func (cc *clusterClient) TTL(ctx context.Context, key string) (time.Duration, error) {
+	result, err := cc.ClusterClient.TTL(ctx, key).Result()
+	if err != nil {
+		return -1, err
+	}
+
+	if result == -2 {
+		return -1, NewKeyNotFoundError(key)
+	}
+
+	if result == -1 {
+		return -1, NewTTLNotFoundError(key)
+	}
+
+	return result, nil
+}
+
+// ZAdd call redis ZADD function
+func (cc *clusterClient) ZAdd(ctx context.Context, key, member string, score float64) error {
+	_, err := cc.ClusterClient.ZAdd(ctx, key, &goredis.Z{Score: score, Member: member}).Result()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ZCard call redis ZCARD function
+func (cc *clusterClient) ZCard(ctx context.Context, key string) (int64, error) {
+	result, err := cc.ClusterClient.ZCard(ctx, key).Result()
+	if err != nil {
+		return -1, err
+	}
+
+	if result == 0 {
+		return -1, NewKeyNotFoundError(key)
+	}
+
+	return result, nil
+}
+
+// ZIncrBy call redis ZINCRBY function
+func (cc *clusterClient) ZIncrBy(ctx context.Context, key, member string, increment float64) error {
+	_, err := cc.ClusterClient.ZIncrBy(ctx, key, increment, member).Result()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ZRange call redis ZRANGE function it is inclusive it returns start and stop element
+func (cc *clusterClient) ZRange(ctx context.Context, key string, start, stop int64) ([]*Member, error) {
+	result, err := cc.ClusterClient.ZRangeWithScores(ctx, key, start, stop).Result()
+	if err != nil {
+		return []*Member{}, err
+	}
+
+	var members []*Member = make([]*Member, 0, len(result))
+	for _, member := range result {
+		members = append(members, &Member{
+			Member: fmt.Sprint(member.Member),
+			Score:  member.Score,
+		})
+	}
+
+	return members, nil
+}
+
+// ZRank call redis ZRANK function
+func (cc *clusterClient) ZRank(ctx context.Context, key, member string) (int64, error) {
+	result, err := cc.ClusterClient.ZRank(ctx, key, member).Result()
+	if err != nil {
+		if err.Error() == "redis: nil" {
+			return 0, NewMemberNotFoundError(key)
+		}
+
+		return 0, err
+	}
+
+	return result, nil
+}
+
+// ZRem call redis ZREM function
+func (cc *clusterClient) ZRem(ctx context.Context, key, member string) error {
+	err := cc.ClusterClient.ZRem(ctx, key, member).Err()
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
You are not doing anything with `err`. You can replace this with:
```suggestion
	return cc.ClusterClient.ZRem(ctx, key, member).Err()
```